pkg/hintrunner/zero/zerohint_memcpy.go

```diff
@@ -10,14 +10,23 @@ func createAllocSegmentHinter(resolver hintReferenceResolver) (hinter.Hinter, er
 	return &core.AllocSegment{Dst: hinter.ApCellRef(0)}, nil
 }
 
-func createVMEnterScopeHinter(resolver hintReferenceResolver) (hinter.Hinter, error) {
+func newMemcpyEnterScopeHinter(len hinter.ResOperander) hinter.Hinter {
 	return &GenericZeroHinter{
-		Name: "VMEnterScope",
+		Name: "MemcpyEnterScope",
 		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
-			ctx.ScopeManager.EnterScope(make(map[string]any))
+			//>  vm_enter_scope({'n': ids.len})
+			ctx.ScopeManager.EnterScope(map[string]any{"n": len})
```

Comment:
You are adding ResOperander to the scope. This is only the reference, we need to get the exact value that it holds. As you can see in the memcpy() method, where this hint is used, the len is a Felt, so you have to resolve it as felt and check the error, using ```hinter.ResolveAsFelt(vm, len)```